lib/msf/buffer_control_writer.js

```diff
@@ -0,0 +1,560 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+goog.provide('shaka.msf.BufferControlWriter');
+
+goog.require('shaka.msf.Utils');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.StringUtils');
+
+
+/**
+ * BufferControlWriter class for writing control messages to a buffer
+ * following the draft-11 specification.
+ *
+ * The typical pattern is to instantiate the class and call one of the
+ * marshal methods to write a message to the buffer. The format is always:
+ * wire format type, 16-bit length, message fields, etc.
+ */
+shaka.msf.BufferControlWriter = class {
```

Comment:
Most of this class is really a generic data writer. Maybe we could move it to i.e. `DataViewWriter`, as it could correspond nicely to already existing `DataViewReader`. There are few places where we need to override some data and most of the time we are reinventing the wheel to some extent.